lambda/src/main/scala/pricemigrationengine/handlers/AmendmentHandler.scala

```diff
@@ -44,8 +63,16 @@ object AmendmentHandler extends CohortHandler {
       item: CohortItem
   ): ZIO[Zuora, Failure, SuccessfulAmendmentResult] = {
 
+    def checkStartDate(startDate: LocalDate, subscriptionName: String): ZIO[Any, Failure, Unit] = {
+      if (LocalDate.now().plusDays(startDateLeadTimeRequirement).isAfter(startDate)) ZIO.succeed(())
```

Comment:
Best to avoid relative times as it's very difficult to test. Use zio.Clock instead to get current time.

Reply:
In theory this should be fine, we send notifications, and do the amendments exactly 49 days in advance or whatever the lead time is. But yeah, Kelvin's probably right, if we have any failures with the Notification lambda then this logic would fail. There would probably be a lot of edge cases.

Having a think, all we need to be sure of is that the `startDate` is 49 days after the `whenNotificationSent`. So let's just compare those two fields.

@shtukas